server/resp/src/main/java/org/infinispan/server/resp/commands/tx/EXEC.java

```diff
@@ -72,14 +72,18 @@ private CompletionStage<?> perform(List<TransactionCommand> commands, RespTransa
 
       // Redis has a serializable isolation. Without batching we have read-uncomitted.
       // Using the batching API we have a few more useful features.
-      if (!cache.getCacheConfiguration().invocationBatching().enabled()) {
+      boolean batchEnabled = cache.getCacheConfiguration().invocationBatching().enabled();
+      if (!batchEnabled) {
          log.multiKeyOperationUseBatching();
       } else {
          cache.startBatch();
       }
       Resp3Handler.writeArrayPrefix(commands.size(), curr.allocator());
       return orderlyExecution(next, ctx, commands, 0, CompletableFutures.completedNull())
-            .whenComplete((ignore, t) -> cache.endBatch(true));
+            .thenRun(() -> {
```

Comment:
Is it alright not running this when an exception occurs in the orderlyExecution method?

Reply:
Ah, yeah. I'll revert this.